compiler-core/src/language_server/engine.rs

```diff
@@ -417,6 +415,44 @@ where
 
         completions
     }
+
+    pub fn inlay_hint(&mut self, params: lsp::InlayHintParams) -> Response<Option<Vec<InlayHint>>> {
+        self.respond(|this| {
+            let Some(module) = this.module_for_uri(&params.text_document.uri) else {
+                return Ok(None);
+            };
+            let line_numbers = LineNumbers::new(&module.code);
+
+            let mut hints = vec![];
+            let start =
+                line_numbers.byte_index(params.range.start.line, params.range.start.character);
+            let end = line_numbers.byte_index(params.range.end.line, params.range.end.character);
```

Comment:
How about something like this:

```suggestion
            let selected = line_numbers.src_span(params.range);
```